tasks/release/bump_version.py

```diff
@@ -0,0 +1,75 @@
+"""Helper Commands for making releases
+"""
+
+import os
+
+from .utils import ROOT, build_version, find_version, replace_in_file, Version
+
+from invoke import task
+
+
+def _bump_version_str(version_str, major=False, minor=False, patch=False,
+                      dev=False, final=False):
+    """Actual version bumping logic
+    """
+    old = build_version(version_str)
+
+    # We assume every version bump would mean we enter development for the new
+    # version which is basically the development flow anyway.
+    if major:
+        new = Version(old.major + 1, 0, 0, 0)
+    elif minor:
+        new = Version(old.major, old.minor + 1, 0, 0)
+    elif patch:
+        new = Version(old.major, old.minor, old.patch + 1, 0)
+    elif dev:
+        new_dev = old.dev + 1 if old.dev is not None else 0
+        new = Version(old.major, old.minor, old.patch, new_dev)
+    elif final:
+        new = Version(old.major, old.minor, old.patch, None)
+    else:
+        msg = "Specify one of --major, --minor, --patch, --dev or --final"
+        raise RuntimeError(msg)
+
+    return ".".join(map(str, new[:3])) + (
+        ".dev" + str(new.dev) if new.dev is not None else ""
+    )
+
+# Poor Man's Testing
+assert _bump_version_str("2.0.0.dev0", dev=True) == "2.0.0.dev1"
+assert _bump_version_str("2.0.1.dev1", minor=True) == "2.1.0.dev0"
+assert _bump_version_str("2.1.0.dev1", patch=True) == "2.1.1.dev0"
+assert _bump_version_str("2.1.2.dev1", major=True) == "3.0.0.dev0"
+assert _bump_version_str("3.0.0.dev1", final=True) == "3.0.0"
+assert _bump_version_str("3.0.0", dev=True) == "3.0.0.dev0"
+
+
+@task(help={
+    'major': "Bumps the major segment.",
+    'minor': "Bumps the minor segment, retaining the major segment.",
+    'patch': "Bumps the patch segment, retaining the major and minor segment.",
+    'dev': "Bumps the dev segment, retaining the other segments.",
+    'final': "Drops any dev segment in the version.",
+})
+def bump_version(ctx, major=False, minor=False, patch=False,
```

Comment:
We could use `zest.releaser`'s `bumpversion` command for that part.

Reply:
FWIW, I do think this should use some existing tool, I'm not sure which one though.

I'll look at `zest.releaser` soon. :)